From 5422b0241f8c3473c41258a8a86acd963f9857f8 Mon Sep 17 00:00:00 2001
From: Andrew Liu <shengping.liu@windriver.com>
Date: Sun, 19 Sep 2010 15:27:16 +0800
Subject: [PATCH 23/26] dts:convert p4080 dev trees to 4 interrupt cells

Patch taken from FSL vendor SDK 2.1.

-in order to support error interrupts we now
     have 4 cells

-update error interrupt specifier definitions
     for mem ctrl, pme, pci, and srio

-srio driver process the error interrupt. if the error
interrupt moves away,the sequence of interrupt register
will be broken.

-remove the error interrupt as it's not appropriately
supported by all the drivers

Signed-off-by: Stuart Yoder <stuart.yoder@freescale.com>
Signed-off-by: Roy Zang <tie-fei.zang@freescale.com>
Signed-off-by: Lan Chunhe-B25806 <b25806@freescale.com>
Signed-off-by: Emil Medve <Emilian.Medve@Freescale.com>
Integrated-by: Andrew Liu <shengping.liu@windriver.com>
---
 arch/powerpc/boot/dts/p4080ds_36b.dts |  221 +++++++++++++++++----------------
 1 files changed, 114 insertions(+), 107 deletions(-)

diff --git a/arch/powerpc/boot/dts/p4080ds_36b.dts b/arch/powerpc/boot/dts/p4080ds_36b.dts
index 973a316..23a7ee5 100644
--- a/arch/powerpc/boot/dts/p4080ds_36b.dts
+++ b/arch/powerpc/boot/dts/p4080ds_36b.dts
@@ -186,7 +186,7 @@
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x0 0x4000 0x100000 0x1000>;
 			cpu-handle = <&cpu0>;
-			interrupts = <0x69 2>;
+			interrupts = <0x69 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@4000 {
@@ -194,7 +194,7 @@
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x4000 0x4000 0x101000 0x1000>;
 			cpu-handle = <&cpu1>;
-			interrupts = <0x6b 2>;
+			interrupts = <0x6b 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@8000 {
@@ -202,7 +202,7 @@
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x8000 0x4000 0x102000 0x1000>;
 			cpu-handle = <&cpu2>;
-			interrupts = <0x6d 2>;
+			interrupts = <0x6d 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@c000 {
@@ -210,7 +210,7 @@
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0xc000 0x4000 0x103000 0x1000>;
 			cpu-handle = <&cpu3>;
-			interrupts = <0x6f 2>;
+			interrupts = <0x6f 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@10000 {
@@ -218,7 +218,7 @@
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x10000 0x4000 0x104000 0x1000>;
 			cpu-handle = <&cpu4>;
-			interrupts = <0x71 2>;
+			interrupts = <0x71 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@14000 {
@@ -226,7 +226,7 @@
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x14000 0x4000 0x105000 0x1000>;
 			cpu-handle = <&cpu5>;
-			interrupts = <0x73 2>;
+			interrupts = <0x73 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@18000 {
@@ -234,7 +234,7 @@
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x18000 0x4000 0x106000 0x1000>;
 			cpu-handle = <&cpu6>;
-			interrupts = <0x75 2>;
+			interrupts = <0x75 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@1c000 {
@@ -242,21 +242,21 @@
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x1c000 0x4000 0x107000 0x1000>;
 			cpu-handle = <&cpu7>;
-			interrupts = <0x77 2>;
+			interrupts = <0x77 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@20000 {
 			cell-index = <0x8>;
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x20000 0x4000 0x108000 0x1000>;
-			interrupts = <0x79 2>;
+			interrupts = <0x79 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 		bman-portal@24000 {
 			cell-index = <0x9>;
 			compatible = "fsl,p4080-bman-portal", "fsl,bman-portal";
 			reg = <0x24000 0x4000 0x109000 0x1000>;
-			interrupts = <0x7b 2>;
+			interrupts = <0x7b 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -277,7 +277,7 @@
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x0 0x4000 0x100000 0x1000>;
 			cpu-handle = <&cpu0>;
-			interrupts = <0x68 0x2>;
+			interrupts = <0x68 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x0>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -304,7 +304,7 @@
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x4000 0x4000 0x101000 0x1000>;
 			cpu-handle = <&cpu1>;
-			interrupts = <0x6a 0x2>;
+			interrupts = <0x6a 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x1>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -331,7 +331,7 @@
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x8000 0x4000 0x102000 0x1000>;
 			cpu-handle = <&cpu2>;
-			interrupts = <0x6c 0x2>;
+			interrupts = <0x6c 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x2>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -358,7 +358,7 @@
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0xc000 0x4000 0x103000 0x1000>;
 			cpu-handle = <&cpu3>;
-			interrupts = <0x6e 0x2>;
+			interrupts = <0x6e 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x3>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -385,7 +385,7 @@
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x10000 0x4000 0x104000 0x1000>;
 			cpu-handle = <&cpu4>;
-			interrupts = <0x70 0x2>;
+			interrupts = <0x70 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x4>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -412,7 +412,7 @@
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x14000 0x4000 0x105000 0x1000>;
 			cpu-handle = <&cpu5>;
-			interrupts = <0x72 0x2>;
+			interrupts = <0x72 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x5>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -439,7 +439,7 @@
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x18000 0x4000 0x106000 0x1000>;
 			cpu-handle = <&cpu6>;
-			interrupts = <0x74 0x2>;
+			interrupts = <0x74 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x6>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -466,7 +466,7 @@
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x1c000 0x4000 0x107000 0x1000>;
 			cpu-handle = <&cpu7>;
-			interrupts = <0x76 0x2>;
+			interrupts = <0x76 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x7>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -492,7 +492,7 @@
 			cell-index = <0x8>;
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x20000 0x4000 0x108000 0x1000>;
-			interrupts = <0x78 0x2>;
+			interrupts = <0x78 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x8>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -518,7 +518,7 @@
 			cell-index = <0x9>;
 			compatible = "fsl,p4080-qman-portal", "fsl,qman-portal";
 			reg = <0x24000 0x4000 0x109000 0x1000>;
-			interrupts = <0x7a 0x2>;
+			interrupts = <0x7a 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qman-channel-id = <0x9>;
 			fsl,qman-pool-channels = <&qpool1 &qpool2 &qpool3
@@ -639,6 +639,10 @@
 		ranges = <0x00000000 0xf 0xfe000000 0x1000000>;
 		reg = <0xf 0xfe000000 0 0x00001000>;
 
+		soc-sram-error {
+			compatible = "fsl,soc-sram-error";
+		};
+
 		corenet-law {
 			compatible = "fsl,corenet-law";
 			reg = <0x0 0x1000>;
@@ -656,14 +660,12 @@
 			compatible = "fsl,p4080-memory-controller";
 			reg = <0x8000 0x1000>;
 			interrupt-parent = <&mpic>;
-			interrupts = <0x12 2>;
 		};
 
 		memory-controller@9000 {
 			compatible = "fsl,p4080-memory-controller";
 			reg = <0x9000 0x1000>;
 			interrupt-parent = <&mpic>;
-			interrupts = <0x12 2>;
 		};
 
 		L3_0: l3-cache-controller@10000 {
@@ -680,7 +682,7 @@
 		iommu@20000 {
 			compatible = "fsl,p4080-pamu";
 			reg = <0x20000 0x5000>;
-			interrupts = <24 2>;
+			interrupts = <24 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -689,14 +691,14 @@
 			reg = <0x41600 0x200>;
 			msi-available-ranges = <0 0x100>;
 			interrupts = <
-				0xe0 0
-				0xe1 0
-				0xe2 0
-				0xe3 0
-				0xe4 0
-				0xe5 0
-				0xe6 0
-				0xe7 0>;
+				0xe0 0 0 0
+				0xe1 0 0 0
+				0xe2 0 0 0
+				0xe3 0 0 0
+				0xe4 0 0 0
+				0xe5 0 0 0
+				0xe6 0 0 0
+				0xe7 0 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -705,14 +707,14 @@
 			reg = <0x41800 0x200>;
 			msi-available-ranges = <0 0x100>;
 			interrupts = <
-				0xe8 0
-				0xe9 0
-				0xea 0
-				0xeb 0
-				0xec 0
-				0xed 0
-				0xee 0
-				0xef 0>;
+				0xe8 0 0 0
+				0xe9 0 0 0
+				0xea 0 0 0
+				0xeb 0 0 0
+				0xec 0 0 0
+				0xed 0 0 0
+				0xee 0 0 0
+				0xef 0 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -721,14 +723,14 @@
 			reg = <0x41a00 0x200>;
 			msi-available-ranges = <0 0x100>;
 			interrupts = <
-				0xf0 0
-				0xf1 0
-				0xf2 0
-				0xf3 0
-				0xf4 0
-				0xf5 0
-				0xf6 0
-				0xf7 0>;
+				0xf0 0 0 0
+				0xf1 0 0 0
+				0xf2 0 0 0
+				0xf3 0 0 0
+				0xf4 0 0 0
+				0xf5 0 0 0
+				0xf6 0 0 0
+				0xf7 0 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -745,7 +747,7 @@
 				reg = <0x0 0x80>;
 				cell-index = <0>;
 				interrupt-parent = <&mpic>;
-				interrupts = <28 2>;
+				interrupts = <28 2 0 0>;
 			};
 			dma-channel@80 {
 				compatible = "fsl,p4080-dma-channel",
@@ -753,7 +755,7 @@
 				reg = <0x80 0x80>;
 				cell-index = <1>;
 				interrupt-parent = <&mpic>;
-				interrupts = <29 2>;
+				interrupts = <29 2 0 0>;
 			};
 			dma-channel@100 {
 				compatible = "fsl,p4080-dma-channel",
@@ -761,7 +763,7 @@
 				reg = <0x100 0x80>;
 				cell-index = <2>;
 				interrupt-parent = <&mpic>;
-				interrupts = <30 2>;
+				interrupts = <30 2 0 0>;
 			};
 			dma-channel@180 {
 				compatible = "fsl,p4080-dma-channel",
@@ -769,7 +771,7 @@
 				reg = <0x180 0x80>;
 				cell-index = <3>;
 				interrupt-parent = <&mpic>;
-				interrupts = <31 2>;
+				interrupts = <31 2 0 0>;
 			};
 		};
 
@@ -786,7 +788,7 @@
 				reg = <0x0 0x80>;
 				cell-index = <0>;
 				interrupt-parent = <&mpic>;
-				interrupts = <32 2>;
+				interrupts = <32 2 0 0>;
 			};
 			dma-channel@80 {
 				compatible = "fsl,p4080-dma-channel",
@@ -794,7 +796,7 @@
 				reg = <0x80 0x80>;
 				cell-index = <1>;
 				interrupt-parent = <&mpic>;
-				interrupts = <33 2>;
+				interrupts = <33 2 0 0>;
 			};
 			dma-channel@100 {
 				compatible = "fsl,p4080-dma-channel",
@@ -802,7 +804,7 @@
 				reg = <0x100 0x80>;
 				cell-index = <2>;
 				interrupt-parent = <&mpic>;
-				interrupts = <34 2>;
+				interrupts = <34 2 0 0>;
 			};
 			dma-channel@180 {
 				compatible = "fsl,p4080-dma-channel",
@@ -810,7 +812,7 @@
 				reg = <0x180 0x80>;
 				cell-index = <3>;
 				interrupt-parent = <&mpic>;
-				interrupts = <35 2>;
+				interrupts = <35 2 0 0>;
 			};
 		};
 
@@ -820,7 +822,7 @@
 			cell-index = <0>;
 			compatible = "fsl-i2c";
 			reg = <0x118000 0x100>;
-			interrupts = <38 2>;
+			interrupts = <38 2 0 0>;
 			interrupt-parent = <&mpic>;
 			dfsrr;
 		};
@@ -831,7 +833,7 @@
 			cell-index = <1>;
 			compatible = "fsl-i2c";
 			reg = <0x118100 0x100>;
-			interrupts = <38 2>;
+			interrupts = <38 2 0  0 0>;
 			interrupt-parent = <&mpic>;
 			dfsrr;
 			eeprom@51 {
@@ -845,7 +847,7 @@
 			rtc@68 {
 				compatible = "dallas,ds3232";
 				reg = <0x68>;
-				interrupts = <0 0x1>;
+				interrupts = <0 0x1 0 0>;
 				interrupt-parent = <&mpic>;
 			};
 		};
@@ -856,7 +858,7 @@
 			cell-index = <2>;
 			compatible = "fsl-i2c";
 			reg = <0x119000 0x100>;
-			interrupts = <39 2>;
+			interrupts = <39 2 0 0>;
 			interrupt-parent = <&mpic>;
 			dfsrr;
 		};
@@ -867,7 +869,7 @@
 			cell-index = <3>;
 			compatible = "fsl-i2c";
 			reg = <0x119100 0x100>;
-			interrupts = <39 2>;
+			interrupts = <39 2 0 0>;
 			interrupt-parent = <&mpic>;
 			dfsrr;
 		};
@@ -878,7 +880,7 @@
 			#size-cells = <0>;
 			compatible = "fsl,espi";
 			reg = <0x110000 0x1000>;
-			interrupts = <53 0x2>;
+			interrupts = <53 0x2 0 0>;
 			interrupt-parent = <&mpic>;
 			espi,num-ss-bits = <4>;
 			mode = "cpu";
@@ -919,7 +921,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			interrupt-parent = <&mpic>;
-			interrupts = <44 0x2>;
+			interrupts = <44 0x2 0 0>;
 			phy_type = "ulpi";
 		};
 
@@ -930,7 +932,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			interrupt-parent = <&mpic>;
-			interrupts = <45 0x2>;
+			interrupts = <45 0x2 0 0>;
 			dr_mode = "host";
 			phy_type = "ulpi";
 		};
@@ -941,7 +943,7 @@
 			compatible = "ns16550";
 			reg = <0x11c500 0x100>;
 			clock-frequency = <0>;
-			interrupts = <36 2>;
+			interrupts = <36 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -951,7 +953,7 @@
 			compatible = "ns16550";
 			reg = <0x11c600 0x100>;
 			clock-frequency = <0>;
-			interrupts = <36 2>;
+			interrupts = <36 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -961,7 +963,7 @@
 			compatible = "ns16550";
 			reg = <0x11d500 0x100>;
 			clock-frequency = <0>;
-			interrupts = <37 2>;
+			interrupts = <37 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -971,7 +973,7 @@
 			compatible = "ns16550";
 			reg = <0x11d600 0x100>;
 			clock-frequency = <0>;
-			interrupts = <37 2>;
+			interrupts = <37 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -1005,14 +1007,18 @@
 			clock-frequency = <0>;
 			interrupt-controller;
 			#address-cells = <0>;
-			#interrupt-cells = <2>;
+			#interrupt-cells = <4>;
 			reg = <0x40000 0x40000>;
 			compatible = "chrp,open-pic";
 			device_type = "open-pic";
 			big-endian;
 			message@1400 {
 				compatible = "fsl,mpic-msg";
-				interrupts = <0xb0 2 0xb1 2 0xb2 2 0xb3 2>;
+				interrupts = <
+                                        0xb0 2 0 0
+                                        0xb1 2 0 0
+                                        0xb2 2 0 0
+                                        0xb3 2 0 0>;
 				interrupt-parent = < &mpic >;
 			};
 		};
@@ -1020,7 +1026,7 @@
 		sdhc: sdhc@114000 {
 			compatible = "fsl,p4080-esdhc", "fsl,esdhc";
 			reg = <0x114000 0x1000>;
-			interrupts = <48 2>;
+			interrupts = <48 2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,sdhci-auto-cmd12;
 			clock-frequency = <0>;
@@ -1037,35 +1043,35 @@
 			#size-cells = <1>;
 			reg		 = <0x300000 0x10000>;
 			ranges		 = <0 0x300000 0x1000>;
-			interrupts	 = <92 2>;
+			interrupts	 = <92 2 0 0>;
 			interrupt-parent = <&mpic>;
 			fsl,qi-spids	 = <5>;
 
 			sec_jq0: jq@1000 {
 				compatible = "fsl,p4080-sec4.0-job-queue", "fsl,sec4.0-job-queue";
 				reg	   = <0x1000 0x1000>;
-				interrupts	 = <88 2>;
+				interrupts	 = <88 2 0 0>;
 				interrupt-parent = <&mpic>;
 			};
 
 			sec_jq1: jq@2000 {
 				compatible = "fsl,p4080-sec4.0-job-queue", "fsl,sec4.0-job-queue";
 				reg	   = <0x2000 0x1000>;
-				interrupts	 = <89 2>;
+				interrupts	 = <89 2 0 0>;
 				interrupt-parent = <&mpic>;
 			};
 
 			sec_jq2: jq@3000 {
 				compatible = "fsl,p4080-sec4.0-job-queue", "fsl,sec4.0-job-queue";
 				reg	   = <0x3000 0x1000>;
-				interrupts	 = <90 2>;
+				interrupts	 = <90 2 0 0>;
 				interrupt-parent = <&mpic>;
 			};
 
 			sec_jq3: jq@4000 {
 				compatible = "fsl,p4080-sec4.0-job-queue", "fsl,sec4.0-job-queue";
 				reg	   = <0x4000 0x1000>;
-				interrupts	 = <91 2>;
+				interrupts	 = <91 2 0 0>;
 				interrupt-parent = <&mpic>;
 			};
 
@@ -1101,7 +1107,7 @@
 		snvs: snvs@314000 {
 			compatible = "fsl,p4080-sec4.0-snvs", "fsl,sec4.0-snvs";
 			reg	   = <0x314000 0x1000>;
-			interrupts	 = <93 2>;
+			interrupts	 = <93 2 0 0>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -1110,7 +1116,6 @@
 			reg = <0x316000 0x10000>;
 			/* fsl,pme-pdsr = <0x0 0x23000000 0x0 0x01000000>; */
 			/* fsl,pme-sre = <0x0 0x24000000 0x0 0x00a00000>; */
-			interrupts = <16 2>;
 			interrupt-parent = <&mpic>;
 		};
 
@@ -1137,7 +1142,7 @@
 			ranges = <0 0x400000 0x100000>;
 			reg = <0x400000 0x100000>;
 			clock-frequency = <666>;
-			interrupts = <96 2>;
+			interrupts = <96 2 0 0>;
 			interrupt-parent = <&mpic>;
 
 			cc@0 {
@@ -1289,7 +1294,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-mdio";
 				reg = <0xe1120 0xee0>;
-				interrupts = <100 1>;
+				interrupts = <100 1 0 0>;
 
 				tbi0: tbi-phy@8 {
 					reg = <0x8>;
@@ -1388,7 +1393,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-tbi";
 				reg = <0xe3120 0xee0>;
-				interrupts = <100 1>;
+				interrupts = <100 1 0 0>;
 
 				tbi1: tbi-phy@8 {
 					reg = <8>;
@@ -1411,7 +1416,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-tbi";
 				reg = <0xe5120 0xee0>;
-				interrupts = <100 1>;
+				interrupts = <100 1 0 0>;
 
 				tbi2: tbi-phy@8 {
 					reg = <8>;
@@ -1434,7 +1439,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-tbi";
 				reg = <0xe7120 0xee0>;
-				interrupts = <100 1>;
+				interrupts = <100 1 0 0>;
 
 				tbi3: tbi-phy@8 {
 					reg = <8>;
@@ -1455,7 +1460,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-xmdio";
 				reg = <0xf1000 0x1000>;
-				interrupts = <100 1>;
+				interrupts = <100 1 0 0>;
 
 				p4080xmdio1: p4080ds-xmdio1 {
 					#address-cells = <1>;
@@ -1491,7 +1496,7 @@
 			ranges = <0 0x500000 0x100000>;
 			reg = <0x500000 0x100000>;
 			clock-frequency = <666>;
-			interrupts = <97 2>;
+			interrupts = <97 2 0 0>;
 			interrupt-parent = <&mpic>;
 
 			cc@0 {
@@ -1642,7 +1647,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-tbi";
 				reg = <0xe1120 0xee0>;
-				interrupts = <101 1>;
+				interrupts = <101 1 0 0>;
 
 				tbi5: tbi-phy@8 {
 					reg = <8>;
@@ -1665,7 +1670,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-tbi";
 				reg = <0xe3120 0xee0>;
-				interrupts = <101 1>;
+				interrupts = <101 1 0 0>;
 
 				tbi6: tbi-phy@8 {
 					reg = <8>;
@@ -1688,7 +1693,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-tbi";
 				reg = <0xe5120 0xee0>;
-				interrupts = <101 1>;
+				interrupts = <101 1 0 0>;
 
 				tbi7: tbi-phy@8 {
 					reg = <8>;
@@ -1711,7 +1716,7 @@
 				#size-cells = <0>;
 				compatible = "fsl,fman-tbi";
 				reg = <0xe7120 0xee0>;
-				interrupts = <101 1>;
+				interrupts = <101 1 0 0>;
 
 				tbi8: tbi-phy@8 {
 					reg = <8>;
@@ -1737,9 +1742,14 @@
 		reg = <0xf 0xfe0c0000 0 0x20000>;
 		ranges = <0 0 0xc 0x20000000 0 0x01000000>;
 		interrupt-parent = <&mpic>;
-		/* err_irq bell_outb_irq bell_inb_irq
-			msg1_tx_irq msg1_rx_irq	msg2_tx_irq msg2_rx_irq */
-		interrupts = <16 2 56 2 57 2 60 2 61 2 62 2 63 2>;
+		interrupts = <
+			16 2 1 11 /* err_irq */
+                        56 2 0 0  /* bell_outb_irq */
+                        57 2 0 0  /* bell_inb_irq */
+                        60 2 0 0  /* msg1_tx_irq */
+                        61 2 0 0  /* msg1_rx_irq */
+                        62 2 0 0  /* msg2_tx_irq */
+                        63 2 0 0>; /* msg2_rx_irq */
 	};
 
 	pci0: pcie@ffe200000 {
@@ -1755,15 +1765,14 @@
 		clock-frequency = <0x1fca055>;
 		fsl,msi = <&msi0>;
 		interrupt-parent = <&mpic>;
-		interrupts = <16 2>;
 
 		interrupt-map-mask = <0xf800 0 0 7>;
 		interrupt-map = <
 			/* IDSEL 0x0 */
-			0000 0 0 1 &mpic 40 1
-			0000 0 0 2 &mpic 1 1
-			0000 0 0 3 &mpic 2 1
-			0000 0 0 4 &mpic 3 1
+			0000 0 0 1 &mpic 40 1 0 0
+			0000 0 0 2 &mpic 1 1 0 0
+			0000 0 0 3 &mpic 2 1 0 0
+			0000 0 0 4 &mpic 3 1 0 0
 			>;
 		pcie@0 {
 			reg = <0 0 0 0 0>;
@@ -1793,14 +1802,13 @@
 		clock-frequency = <0x1fca055>;
 		fsl,msi = <&msi1>;
 		interrupt-parent = <&mpic>;
-		interrupts = <16 2>;
 		interrupt-map-mask = <0xf800 0 0 7>;
 		interrupt-map = <
 			/* IDSEL 0x0 */
-			0000 0 0 1 &mpic 41 1
-			0000 0 0 2 &mpic 5 1
-			0000 0 0 3 &mpic 6 1
-			0000 0 0 4 &mpic 7 1
+			0000 0 0 1 &mpic 41 1 0 0
+			0000 0 0 2 &mpic 5 1 0 0
+			0000 0 0 3 &mpic 6 1 0 0
+			0000 0 0 4 &mpic 7 1 0 0
 			>;
 		pcie@0 {
 			reg = <0 0 0 0 0>;
@@ -1830,14 +1838,13 @@
 		clock-frequency = <0x1fca055>;
 		fsl,msi = <&msi2>;
 		interrupt-parent = <&mpic>;
-		interrupts = <16 2>;
 		interrupt-map-mask = <0xf800 0 0 7>;
 		interrupt-map = <
 			/* IDSEL 0x0 */
-			0000 0 0 1 &mpic 42 1
-			0000 0 0 2 &mpic 9 1
-			0000 0 0 3 &mpic 10 1
-			0000 0 0 4 &mpic 11 1
+			0000 0 0 1 &mpic 42 1 0 0
+			0000 0 0 2 &mpic 9  1 0 0
+			0000 0 0 3 &mpic 10 1 0 0
+			0000 0 0 4 &mpic 11 1 0 0
 			>;
 		pcie@0 {
 			reg = <0 0 0 0 0>;
@@ -1912,7 +1919,7 @@
 	localbus@ffe124000 {
 		compatible = "fsl,p4080-elbc", "fsl,elbc", "simple-bus";
 		reg = <0xf 0xfe124000 0 0x1000>;
-		interrupts = <25 2>;
+		interrupts = <25 2 0 0>;
 		interrupt-parent = <&mpic>;
 		#address-cells = <2>;
 		#size-cells = <1>;
-- 
1.6.5.2

